homeassistant/components/nextdns/select.py

```diff
@@ -0,0 +1,131 @@
+"""Support for the NextDNS service."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Generic
+
+from aiohttp import ClientError
+from aiohttp.client_exceptions import ClientConnectorError
+from nextdns import ApiError, Settings
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import CoordinatorDataT, NextDnsSettingsUpdateCoordinator
+from .const import ATTR_SETTINGS, DOMAIN
+
+PARALLEL_UPDATES = 1
+
+
+RETENTION_MAP = {
+    "one_hour": 1,
+    "six_hours": 6,
+    "one_day": 24,
+    "one_week": 168,
+    "one_month": 720,
+    "three_months": 2160,
+    "six_months": 4320,
+    "one_year": 8760,
+    "two_years": 17520,
+}
+RETENTION_INVERTED_MAP = {v: k for k, v in RETENTION_MAP.items()}
+
+
+@dataclass
+class NextDnsSelectRequiredKeysMixin(Generic[CoordinatorDataT]):
+    """Class for NextDNS entity required keys."""
+
+    current_option: Callable[[CoordinatorDataT], str]
+    select_option_method: str
+    option_map: dict[str, Any]
+
+
+@dataclass
+class NextDnsSelectEntityDescription(
+    SelectEntityDescription, NextDnsSelectRequiredKeysMixin[CoordinatorDataT]
+):
+    """NextDNS select entity description."""
+
+
+LOGS_RETENTION_SELECT = NextDnsSelectEntityDescription[Settings](
+    key="logs_retention",
+    name="Logs retention",
```

Comment:
Not to complain about everything in this PR, but isn't this one similar useful to log location. You configure it once and then never again ?

Reply:
I change retention during use but agree that this is rarely done.

Reply:
On the other hand, it happens that everyone says it is not useful and suddenly there is a user who says I use it all the time. We cannot imagine all possible use cases 😀

Reply:
Very true. But we don't have to cover all possible use cases either. We shouldn't be adding and maintaining code because 1 user wants it.

Reply:
My heart bleeds, but I accept your arguments, I close PR.